firehose/firehose.go

```diff
@@ -2,25 +2,72 @@ package firehose
 
 import (
 	"crypto/tls"
-	"fmt"
+	caching "github.com/cloudfoundry-community/firehose-to-syslog/caching"
+	log "github.com/cloudfoundry-community/firehose-to-syslog/logging"
 	"github.com/cloudfoundry/noaa"
 	"github.com/cloudfoundry/noaa/events"
-	"os"
+	"fmt"
 )
 
 func CreateFirehoseChan(DopplerEndpoint string, Token string, subId string, skipSSLValidation bool) chan *events.Envelope {
 	connection := noaa.NewConsumer(DopplerEndpoint, &tls.Config{InsecureSkipVerify: skipSSLValidation}, nil)
+
+	connection.SetDebugPrinter(ConsoleDebugPrinter{})
+
 	msgChan := make(chan *events.Envelope)
 	go func() {
 		errorChan := make(chan error)
 		defer close(msgChan)
-		defer close(errorChan)
+
+		defer func() {
+			if r := recover(); r != nil {
+				log.LogError("Recovered in CreateFirehoseChan Thread!", r)
+			}
+		}()
 
 		go connection.Firehose(subId, Token, msgChan, errorChan, nil)
 
 		for err := range errorChan {
-			fmt.Fprintf(os.Stderr, "%v\n", err.Error())
+			log.LogError("Firehose Error!", err.Error())
 		}
 	}()
 	return msgChan
 }
+
+func GetAppContainerMetrics(DopplerEndpoint string, Token string, app caching.App, skipSSLValidation bool) []*events.Envelope {
```

Comment:
Not sure this one is required... Everything should come from the firehose, let s say we have thousand of apps ...